src/alchemlyb/parsing/util.py

```diff
@@ -10,41 +12,71 @@ def bz2_open(filename, mode):
     mode += 't' if mode in ['r','w','a','x'] else ''
     return bz2.open(filename, mode)
 
+
 def gzip_open(filename, mode):
     mode += 't' if mode in ['r','w','a','x'] else ''
     return gzip.open(filename, mode)
 
-def anyopen(filename, mode='r'):
-    """Return a file stream for filename, even if compressed.
 
-    Supports files compressed with bzip2 (.bz2), gzip (.gz), and zip (.zip)
-    compression schemes. The appropriate extension must be present for
-    the function to properly handle the file.
+def anyopen(datafile: Union[PathLike, IO], mode='r', compression=None):
+    """Return a file stream for file or stream, even if compressed.
+
+    Supports files compressed with bzip2 (.bz2) and gzip (.gz) compression
+    schemes. The appropriate extension must be present for the function to
+    properly handle the file.
+
+    If giving a stream for `datafile`, then you must specify `compression` if
+    the stream is compressed.
 
     Parameters
     ----------
-    filename : str
-        Path to file to use.
+    datafile : PathLike | IO
+        Path to file to use, or an open IO stream. If an IO stream, use
+        `compression` to specify the type of compression used, if any.
     mode : str
         Mode for stream; usually 'r' or 'w'.
+    compression : str
+        Use to specify compression. Must be one of 'bzip2', 'gzip'.
+        Overrides use of extension for determining compression if `datafile` is
+        a file.
 
     Returns
     -------
     stream : stream
-        Open stream for reading.
+        Open stream for reading or writing, depending on mode.
 
     """
     # opener for each type of file
     extensions = {'.bz2': bz2_open,
-                  '.gz': gzip_open,
-                  '.zip': zipfile.ZipFile}
```

Comment:
Why did you remove zip?

Reply:
... and remove `import zipfile`

Reply:
Zip doesn't work like `gzip` or `bz2` because it's an archive format, not merely a compressor. Reading a "zip-compressed" file requires specifying the file's name in the archive as well. 

The functionality we intended to have here likely never worked (mid-twenties me is to blame :sweat_smile:), so I'm removing it. We don't have test coverage of it, either, nor any `zip`-archived files in `alchemtest`.

Reply:
Addressed the above items!

Reply:
Thanks for the explanation!